functions/sign_string_rs256_pkcs8_test.go

```diff
@@ -0,0 +1,111 @@
+package functions
+
+import (
+	"testing"
+)
+
+func TestFormatKey(t *testing.T) {
+	unencrypted := `-----BEGIN PRIVATE KEY----- abcdef ghijkl mnopqr -----END PRIVATE KEY-----`
+	unencryptedExpected := `-----BEGIN PRIVATE KEY-----
+abcdef
+ghijkl
+mnopqr
+-----END PRIVATE KEY-----`
+
+	encrypted := `-----BEGIN ENCRYPTED PRIVATE KEY----- abcdef ghijkl mnopqr -----END ENCRYPTED PRIVATE KEY-----`
+	encryptedExpected := `-----BEGIN ENCRYPTED PRIVATE KEY-----
+abcdef
+ghijkl
+mnopqr
+-----END ENCRYPTED PRIVATE KEY-----`
+
+	var FormatKeyTests = []struct {
+		input     string
+		encrypted bool
+		expected  string
+		err       error
+	}{
+		{unencrypted, false, unencryptedExpected, nil},
+		{encrypted, true, encryptedExpected, nil},
+		{unencrypted, true, "", errInvalidKeyFormat},
+		{encrypted, false, "", errInvalidKeyFormat},
+	}
+
+	for _, tc := range FormatKeyTests {
+		actual, err := formatKey(tc.input, tc.encrypted)
+		if actual != tc.expected {
+			t.Errorf("formatKey(%v, %v): expected %v, actual %v", tc.input, tc.encrypted, tc.expected, actual)
+		}
+		if err != tc.err {
+			t.Errorf("formatKey(%v, %v): expected %v, got: %v", tc.input, tc.encrypted, tc.err, err)
+		}
+	}
+}
+
+func TestValidateSignStringRS256PKCS8(t *testing.T) {
+	var tests = []struct {
+		args  []string
+		valid bool
+	}{
+		{[]string{}, false},
+		{[]string{"key"}, false},
+		{[]string{"key", "passphrase"}, false},
+		{[]string{"key", ""}, false},
+		{[]string{"key", "passphrase", "string"}, true},
+		{[]string{"key", "", "string"}, true},
```

Comment:
```suggestion
		{[]string{"key", "", "string"}, true},
		{[]string{"key", "passphrase", "string", "string"}, true},
		{[]string{"key", "", "string", "string", "string", "string"}, true},
```
A couple of tests for multi-arg strings.